oasislmf/pytools/gulmc/manager.py

```diff
@@ -0,0 +1,886 @@
+          number of bins populated for the effective damageability function in `eff_vuln_cdf`.
+    """
+    # init data structures
+    haz_prob_start_in_footprint = List.empty_list(nb_int64)
+    areaperil_ids = List.empty_list(nb_areaperil_int)
+
+    footprint_i = 0
+    last_areaperil_id = nb_areaperil_int(0)
+    last_areaperil_id_start = nb_int64(0)
+    haz_cdf_i = nb_int64(0)
+    areaperil_to_haz_cdf = Dict.empty(nb_areaperil_int, nb_int64)
+
+    Nevent_footprint_entries = len(event_footprint)
+    haz_pdf = np.empty(Nevent_footprint_entries, dtype=oasis_float)  # max size
+    haz_cdf = np.empty(Nevent_footprint_entries, dtype=oasis_float)  # max size
+    Nvulns, Ndamage_bins_max, Nint_bins = vuln_array.shape
+
+    eff_vuln_cdf = np.empty((Nvulns * Ndamage_bins_max), dtype=oasis_float)  # max size
+    cdf_start = 0
+    cdf_end = 0
+    haz_cdf_ptr = List([0])
+    eff_vuln_cdf_start = 0
+    areaperil_to_eff_vuln_cdf = Dict.empty(ITEM_MAP_KEY_TYPE_internal, nb_int64)
+    areaperil_to_eff_vuln_cdf_Ndamage_bins = Dict.empty(ITEM_MAP_KEY_TYPE_internal, nb_int64)
+
+    while footprint_i < Nevent_footprint_entries:
+
+        areaperil_id = event_footprint[footprint_i]['areaperil_id']
+
+        if areaperil_id != last_areaperil_id:
+            # one areaperil_id is completed
+
+            if last_areaperil_id > 0:
+                if last_areaperil_id in areaperil_to_vulns_idx_dict:
+                    # if items with this areaperil_id exist, process and store this areaperil_id
+                    areaperil_ids.append(last_areaperil_id)
+                    haz_prob_start_in_footprint.append(last_areaperil_id_start)
+                    areaperil_to_haz_cdf[last_areaperil_id] = haz_cdf_i
+                    haz_cdf_i += 1
+
+                    # read the hazard intensity pdf and compute the cdf
+                    Nbins_to_read = footprint_i - last_areaperil_id_start
+                    cdf_end = cdf_start + Nbins_to_read
+                    cumsum = 0
+                    for i in range(Nbins_to_read):
+                        haz_pdf[cdf_start + i] = event_footprint['probability'][last_areaperil_id_start + i]
+                        cumsum += haz_pdf[cdf_start + i]
+                        haz_cdf[cdf_start + i] = cumsum
+
+                    # compute effective damageability cdf (internally called `eff_vuln`)
+                    # for each vulnerability function associated to this areaperil_id, compute the product between
+                    # the hazard intensity probability and the damage probability.
+                    areaperil_to_vulns_idx = areaperil_to_vulns_idx_array[areaperil_to_vulns_idx_dict[last_areaperil_id]]
+                    for vuln_idx in range(areaperil_to_vulns_idx['start'], areaperil_to_vulns_idx['end']):
+                        eff_vuln_cdf_cumsum = 0.
+                        Ndamage_bins = 0
```

Comment:
should the name Ndamage_bins be more damage_bin_i, even if at the end it will represent the number of damage_bin

Reply:
I agree it is counter-intuitive as the name `Ndamage_bins` should typically indicate a total number of bins. 
For efficiency purposes, I am using it as the loop counter, and then I use its final value to avoid setting another variable. 
However, if you think that this would be better I can change it to this:
```py
bin_i = 0
while bin_i < Ndamage_bins_max:
    for i in range(Nbins_to_read):
        intensity_bin_i = event_footprint['intensity_bin_id'][last_areaperil_id_start + i] - 1
        eff_vuln_cdf_cumsum += vuln_array[vuln_idx,
                                          bin_i, intensity_bin_i] * haz_pdf[cdf_start + i]
    eff_vuln_cdf[eff_vuln_cdf_start + bin_i] = eff_vuln_cdf_cumsum
    bin_i += 1
    if eff_vuln_cdf_cumsum > 0.999999940:
              break
Ndamage_bins = bin_i
```

Reply:
why not damage_bin_i as we have multiple variables that are represented by bin?

Reply:
fine with `damage_bin_i`, I'll change to that

Reply:
done, I've changed this in two other points of the code.